app/forms/idv/finance_form.rb

```diff
@@ -0,0 +1,46 @@
+module Idv
+  class FinanceForm
+    include ActiveModel::Model
+
+    FINANCE_PARAMS = [:ccn, :mortgage, :home_equity_line, :auto_loan].freeze
+
+    attr_reader :idv_params, :finance_type, :finance_account
+
+    validates :finance_type, :finance_account, presence: true
+
+    validate :finance_type_valid
+
+    def initialize(idv_params)
+      @idv_params = idv_params
+      @finance_type = FINANCE_PARAMS.find { |param| idv_params.key? param }
+      @finance_account = idv_params[@finance_type]
+    end
+
+    def submit(params)
+      @finance_type = params[:finance_type]
+      @finance_account = params[:finance_account]
+      return false unless valid?
+      clear_idv_params_finance
```

Comment:
I see. Can we write a test for this?

How does the proofer gem parse this hash? What happens if there are 2 finance types?